timeseries/src/autogluon/timeseries/models/ensemble/array_based/regressor/linear_stacker.py

```diff
@@ -0,0 +1,180 @@
+        self.tolerance_change = tolerance_change
+        self.tolerance_grad = tolerance_grad
+
+        # Learned weights (stored as numpy arrays)
+        self.weights: Optional[np.ndarray] = None
+
+    def _compute_weight_shape(self, base_model_predictions_shape: tuple) -> tuple:
+        """Compute weight tensor shape based on weights_per configuration."""
+        _, _, prediction_length, num_outputs, num_models = base_model_predictions_shape
+
+        shapes = {
+            "m": (1, 1, num_models),
+            "mt": (prediction_length, 1, num_models),
+            "mq": (1, num_outputs, num_models),
+            "mtq": (prediction_length, num_outputs, num_models),
+        }
+        try:
+            return (1, 1) + shapes[self.weights_per]
+        except KeyError:
+            raise ValueError(f"Unsupported weights_per: {self.weights_per}")
+
+    def make_weighted_average_module(self, base_model_predictions_shape: tuple):
+        import torch
+
+        class WeightedAverage(torch.nn.Module):
+            def __init__(self, shape):
+                super().__init__()
+                self.raw_weights = torch.nn.Parameter(torch.zeros(*shape, dtype=torch.float64))
+
+            def get_normalized_weights(self):
+                return torch.softmax(self.raw_weights, dim=-1)  # softmax over models
+
+            def forward(self, base_model_predictions: torch.Tensor):
+                return torch.sum(self.get_normalized_weights() * base_model_predictions, dim=-1)
+
+        return WeightedAverage(self._compute_weight_shape(base_model_predictions_shape))
+
+    def fit(
+        self,
+        base_model_mean_predictions: np.ndarray,
+        base_model_quantile_predictions: np.ndarray,
+        labels: np.ndarray,
+        time_limit: Optional[float] = None,
+    ) -> Self:
+        import torch
+
+        def _wql(
+            labels_tensor: torch.Tensor,
+            ensemble_predictions: torch.Tensor,
+        ) -> torch.Tensor:
+            """Compute the weighted quantile loss on predictions and ground truth (labels).
+            Considering that the first dimension of predictions is the mean, we treat
+            mean predictions on the same footing as median (0.5) predictions as contribution
+            to the overall weighted quantile loss.
+            """
+            quantile_levels = torch.tensor([0.5] + self.quantile_levels, dtype=torch.float64)
+            error = labels_tensor - ensemble_predictions  # (num_windows, num_items, num_time, num_outputs)
+            quantile_loss = torch.maximum(quantile_levels * error, (quantile_levels - 1) * error)
+            return torch.sum(quantile_loss, dim=-1).mean()
+
+        timer = Timer(time_limit).start()
+
+        base_model_predictions = torch.tensor(
+            np.concatenate(
+                [base_model_mean_predictions, base_model_quantile_predictions],
+                axis=3,
+            ),
+            dtype=torch.float64,
```

Comment:
Let's use float32 everywhere instead of float64? I don't think there is any benefit for using double here.